core/dbt/events/functions.py

```diff
@@ -1,111 +1,129 @@
 import betterproto
-from colorama import Style
-from dbt.events.base_types import NoStdOut, BaseEvent, NoFile, Cache
-from dbt.events.types import EventBufferFull, MainReportVersion, EmptyLine
-import dbt.flags as flags
 from dbt.constants import SECRET_ENV_PREFIX, METADATA_ENV_PREFIX
-
-from dbt.logger import make_log_dir_if_missing, GLOBAL_LOGGER
-from datetime import datetime
+from dbt.events.base_types import BaseEvent, Cache, EventLevel, NoFile, NoStdOut
+from dbt.events.eventmgr import EventManager, LoggerConfig, LineFormat, NoFilter
+from dbt.events.types import EmptyLine
+import dbt.flags as flags
+from dbt.logger import GLOBAL_LOGGER, make_log_dir_if_missing
+from functools import partial
 import json
-import io
-from io import StringIO, TextIOWrapper
-import logbook
-import logging
-from logging import Logger
-import sys
-from logging.handlers import RotatingFileHandler
 import os
+import sys
+from typing import Callable, Dict, List, Optional, TextIO
 import uuid
-import threading
-from typing import List, Optional, Union, Callable, Dict
-from collections import deque
+
 
 LOG_VERSION = 3
-EVENT_HISTORY = None
-
-# create the global file logger with no configuration
-FILE_LOG = logging.getLogger("default_file")
-null_handler = logging.NullHandler()
-FILE_LOG.addHandler(null_handler)
-
-# set up logger to go to stdout with defaults
-# setup_event_logger will be called once args have been parsed
-STDOUT_LOG = logging.getLogger("default_stdout")
-STDOUT_LOG.setLevel(logging.INFO)
-stdout_handler = logging.StreamHandler(sys.stdout)
-stdout_handler.setLevel(logging.INFO)
-STDOUT_LOG.addHandler(stdout_handler)
-
-format_color = True
-format_json = False
-invocation_id: Optional[str] = None
 metadata_vars: Optional[Dict[str, str]] = None
 
+# The default event manager will not log anything, but some tests run code that
+# generates events, without configuring the event manager.
+EVENT_MANAGER: EventManager = EventManager()
 
-def setup_event_logger(log_path, level_override=None):
-    global format_json, format_color, STDOUT_LOG, FILE_LOG
+
+def setup_event_logger(log_path: str, level_override: Optional[EventLevel] = None):
+    cleanup_event_logger()
     make_log_dir_if_missing(log_path)
+    if flags.ENABLE_LEGACY_LOGGER:
```

Comment:
@peterallenwebb This looks great!! Is this the place cloud integrate their desired logger in?

Reply:
No here, but by adding a function to EventManager.callbacks. As you know, we now have a conversation started with @racheldaniel to get the details sorted out around how that callback registration will work.

Reply:
Very happy to see this thread — let me know if/how I can be helpful in that conversation w/ Rachel et al!